### PR TITLE
Update main README.md

### LINKED ISSUE #179: Add documentation for beta release

### LINKED ISSUE #180: Make sure to add migration not supported warning in documentation

### PR DESCRIPTION
**Pull request by Arkatufus**
Fixes #179
Fixes #180

## Changes

* Update and reorganize documentation sections
* Update benchmark table
* Update HOCON section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,29 +6,62 @@ This is a port of the amazing [akka-persistence-jdbc](https://github.com/akka/ak
...
+### This Is Still a Beta
 
-#### This is still a WORK IN PROGRESS
+Please note this is still considered 'work in progress' and only used if one understands the risks. While the TCK Specs pass you should still test in a 'safe' non-production environment carefully before deciding to fully deploy.
 
- **Pull Requests are Welcome** but please note this is still considered 'work in progress' and only used if one understands the risks. While the TCK Specs pass you should still test in a 'safe' non-production environment carefully before deciding to fully deploy.
+### Suitable For Greenfield Projects Only
 
-Working:
+Until backward compatibility is properly tested and documented, it is recommended to use this plugin only on new greenfield projects that does not rely on existing persisted data.
```
Comment (Aaronontheweb) [inline README.md:17]: I think that's pretty clear, LGTM.

File: README.md [hunk 2]
```diff
@@ -88,66 +111,45 @@ Working:
...
-|PersistGroup100 |  607|965| 5537|912.19%|573.78%|
-|PersistGroup200 |  628|1356| 7966|1268.47%|587.46%|
-|PersistGroup25 |  629|675| 2189|348.01%|324.30%|
-|PersistGroup400 |  612|1011| 7237|1182.52%|715.83%|
-|PersistGroup50 |  612|654| 3867|631.86%|591.28%|
-|Recovering |  41903|38766| 42592|101.64%|109.87%|
-|Recovering8 |  75466|65515| 63960|84.75%|97.63%|
-|RecoveringFour |  59259|51355| 58437|98.61%|113.79%|
-|RecoveringTwo |  41745|35512| 41108|98.47%|115.76%|
+| Test            |          SqlServer | SqlServer Batching | Linq2Db | vs Normal | vs Batching |
```
Comment (Aaronontheweb) [inline README.md:129]: The new performance numbers look great - what's responsible for those improvements? Akka.NET v1.5, .NET 7, or a combination of factors?

File: README.md [hunk 3]
```diff
@@ -88,66 +111,45 @@ Working:
...
+| PersistGroup200 |                990 |               1662 |    8086 |   816.77% |     486.52% |
+| PersistGroup25  |               1034 |               1010 |    3054 |   295.36% |     302.38% |
+| PersistGroup400 |               1049 |               2113 |    7237 |   689.59% |     342.50% |
+| PersistGroup50  |                971 |                980 |    4932 |   507.93% |     503.27% |
+| Recovering      |              60516 |              77688 |   64457 |   106.51% |      82.96% |
+| Recovering8     |             116401 |             101549 |  103463 |    88.89% |     101.88% |
+| RecoveringFour  |              86107 |              73218 |   66512 |    77.24% |      90.84% |
+| RecoveringTwo   |              60730 |              53062 |   43325 |    71.34% |      81.65% |
 
 ## Sql.Common Compatibility modes
```
Comment (Aaronontheweb) [inline README.md:146]: I'd say "Akka.Persistence.Sql.Common` here, just to be explicit.

File: README.md [hunk 4]
```diff
@@ -196,23 +196,12 @@ Please note that you -must- provide a Connection String (`connection-string`) an
 
 ```hocon
 akka.persistence {
-  publish-plugin-commands = on
-
   journal {
-    plugin = "akka.persistence.journal.sql"
```
Comment (Aaronontheweb) [inline README.md:202]: Isn't this setting definitely needed in order to get Akka.Persistence to load the plugin correctly?

File: README.md [hunk 5]
```diff
@@ -280,125 +269,433 @@ akka.persistence {
...
 
 Please note that you -must- provide a Connection String and Provider name.
 
-- Refer to the Members of `LinqToDb.ProviderName` for included providers.
-  - Note: For best performance, one should use the most specific provider name possible. i.e. `LinqToDB.ProviderName.SqlServer2012` instead of `LinqToDB.ProviderName.SqlServer`. Otherwise certain provider detections have to run more frequently which may impair performance slightly.
-
 ```hocon
 akka.persistence {
   snapshot-store {
-    plugin = "akka.persistence.snapshot-store.sql"
```
Comment (Aaronontheweb) [inline README.md:379]: Again, isn't this setting needed in order to get this plugin to load correctly?

**Review by Aaronontheweb — COMMENTED** [review]
Left some questions for you @Arkatufus

File: README.md (hunk 2 above)
Comment (Arkatufus) [reply README.md:129]: I really have no idea, all I did was re-run the benchmark on the latest SqlServer and Linq2Db version

File: README.md (hunk 3 above)
Comment (Arkatufus) [reply README.md:146]: It is measured using SqlServer though, we should consider that as a possible variable, there might be optimization at the SQL generator level that might affect benchmarks

File: README.md (hunk 4 above)
Comment (Arkatufus) [reply README.md:202]: No, this is a dump of the default hocon configuration. That line should NOT be in the default configuration file.
It is in the `Setup` section on line 23-38.

File: README.md (hunk 3 above)
Comment (Aaronontheweb) [reply README.md:146]: My comment pertains to the "Sql.Common Compatibility modes" header, not the benchmarks

**Pull request merged**